mypy/checkexpr.py

```diff
@@ -3534,6 +3534,8 @@ def visit_conditional_expr(self, e: ConditionalExpr) -> Type:
 
         if_type = self.analyze_cond_branch(if_map, e.if_expr, context=ctx)
 
+        # Analyze the right branch using full type context and store the type
+        full_context_else_type = self.analyze_cond_branch(else_map, e.else_expr, context=ctx)
         if not mypy.checker.is_valid_inferred_type(if_type):
             # Analyze the right branch disregarding the left branch.
             else_type = self.analyze_cond_branch(else_map, e.else_expr, context=ctx)
```

Comment:
It looks like can replace this with `else_type = full_context_else_type` to avoid making exactly the same call twice.